### PR TITLE
Added missing custom context validation documentation

### PR DESCRIPTION
**Pull request by vijayagrawal18**
### Summary

Added documentation for custom context in Active Record validations.

### CODE REVIEW

**Comment by mechanicles** [comment]
@vijayagrawal18 Use `[ci skip]` in your commit message if a commit is related to the documentation. Keep hacking :smile:

File: guides/source/active_record_validations.md [hunk 1]
```diff
@@ -830,6 +830,22 @@ class Person < ApplicationRecord
...
+`valid?`, `invalid?`, `save`.
+
+```ruby
+class Person < ApplicationRecord
+  validates :email, uniqueness: true, on: :account_setup
+  validates :age, numericality: true, on: :account_setup
+end
+# person = Person.new
+```
+`person.valid?(:account_setup)` exceutes both the validations
```
Comment (Edouard-chin) [inline guides/source/active_record_validations.md]: exceutes -> executes

**Comment by Edouard-chin** [comment]
:+1:

**Comment by vipulnsward** [comment]
LGTM, @vijayagrawal18 can you take care of the typo?

File: guides/source/active_record_validations.md [hunk 2]
```diff
@@ -830,6 +830,22 @@ class Person < ApplicationRecord
...
+Custom contexts need to be triggered explicitly
+by passing name of the context to
+`valid?`, `invalid?`, `save`.
+
+```ruby
+class Person < ApplicationRecord
+  validates :email, uniqueness: true, on: :account_setup
+  validates :age, numericality: true, on: :account_setup
+end
+# person = Person.new
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: This seems out of place.

File: guides/source/active_record_validations.md [hunk 3]
```diff
@@ -830,6 +830,22 @@ class Person < ApplicationRecord
 end
 ```
 
+You can also use `on:` to define custom context.
+Custom contexts need to be triggered explicitly
+by passing name of the context to
+`valid?`, `invalid?`, `save`.
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: `valid?`, `invalid?`, and `save`

File: guides/source/active_record_validations.md (hunk 1 above)
Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: In the case of `person.valid?(:account_setup)`, will it run any other validations other than those from `account_setup` context?

**Comment by vijayagrawal18** [comment]
Thanks guys. @prathamesh-sonpatki Have a look now. That comment is just to make explicit what is `person`. We can leave it.

File: guides/source/active_record_validations.md (hunk 2 above)
Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: @vijayagrawal18 Let's uncomment it then and make it part of the code sample.

File: guides/source/active_record_validations.md [hunk 4]
```diff
@@ -830,6 +830,26 @@ class Person < ApplicationRecord
...
+`valid?`, `invalid?` or `save`.
+
+```ruby
+class Person < ApplicationRecord
+  validates :email, uniqueness: true, on: :account_setup
+  validates :age, numericality: true, on: :account_setup
+end
+# person = Person.new
+```
+`person.valid?(:account_setup)` executes both the validations
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: `will execute` instead of `executes`

File: guides/source/active_record_validations.md [hunk 5]
```diff
@@ -830,6 +830,26 @@ class Person < ApplicationRecord
...
+without saving the model. And `person.save(context: :account_setup)`
+validates `person` in `account_setup` context before saving.
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: `will validate` instead of `validates`

**Comment by prathamesh-sonpatki** [comment]
@vijayagrawal18 Few minor comments. Can you also wrap the lines to 80 characters to match the existing style in the guide?

File: guides/source/active_record_validations.md (hunk 5 above)
Comment (vijayagrawal18) [inline guides/source/active_record_validations.md]: This prefers present tense. http://guides.rubyonrails.org/api_documentation_guidelines.html#wording

Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: :+1:, will keep that in mind :)

**Comment by vijayagrawal18** [comment]
@prathamesh-sonpatki Unwrapped the lines. Please have a look, I will squash it afterwards.

**Comment by prathamesh-sonpatki** [comment]
@vijayagrawal18 Looks good. Please squash

File: guides/source/active_record_validations.md [hunk 6]
```diff
@@ -830,6 +830,24 @@ class Person < ApplicationRecord
...
+by passing name of the context to `valid?`, `invalid?` or `save`.
+
+```ruby
+class Person < ApplicationRecord
+  validates :email, uniqueness: true, on: :account_setup
+  validates :age, numericality: true, on: :account_setup
+end
+
+person = Person.new
+```
```
Comment (prathamesh-sonpatki) [inline guides/source/active_record_validations.md]: Sorry, forgot to mention. Please add a line break after this.

Comment (vijayagrawal18) [inline guides/source/active_record_validations.md]: Done.

**Comment by prathamesh-sonpatki** [comment]
Thanks @vijayagrawal18!

**Pull request merged**